src/isl/util.rs

```diff
@@ -188,6 +191,39 @@ impl Range {
                 // TODO: Implement this section once the timestamp comparator for ion-rust is implemented
                 todo!()
             }
+            Range::TimestampPrecision(start, end) => {
+                let value = &value.as_timestamp().ok_or_else(|| {
+                    invalid_schema_error_raw(
+                        "Timestamp Precision ranges can only have timestamp value for validation",
+                    )
+                })?;
+
+                let value = TimestampPrecisionValue::from_timestamp(value);
+                let is_in_lower_bound = match start {
+                    Min => true,
+                    Value(start_value, boundary_type) => match start_value {
+                        TimestampPrecision(min_value) => match boundary_type {
+                            RangeBoundaryType::Inclusive => min_value <= &value,
+                            RangeBoundaryType::Exclusive => min_value < &value,
+                        },
+                        _ => unreachable!("TimestampPrecision range can only have timestamp precisions as lower and upper range boundary value"),
+                    },
+                    Max => unreachable!("Cannot have 'Max' as the lower range boundary")
```

Comment:
[`unreachable!`](https://doc.rust-lang.org/std/macro.unreachable.html) statements are meant to be used for branches that will never be encountered unless the library has a bug. Is that the case here, or are we using it to validate input? If it's the latter, we should either enforce these requirements earlier in the code (during construction of the values, say) or return an `IonSchemaError` here that explains what they did wrong.

Reply:
These checks are already performed while constructing the `Range` this method is for checking if a range contains the value so all these checks/errors here would be `unreachable!` as they were already verified at the time of range creation.